File: src/main/java/com/poste/ProjetIPM/entities/IPM_Facture.java
package com.poste.ProjetIPM.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class IPM_Facture {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idfacture;
    private String matricule;
    private String code_acte;
    private Integer tarification;
    private Integer part_patient;
    private Integer part_ipm;
    private Double taux_ipm;
    private Date date_facture;
    private Date date_saisie;
    private String montant_facture;

    @OneToMany (mappedBy = "ipm_facture")
    private List<IPM_Bon> ipm_bons;

    @ManyToOne
    @JoinColumn(name = "code_prestataire")
    private IPM_Prestataire ipm_prestataire;

    @OneToMany(mappedBy = "ipm_facture")
    private List<IPM_Statut_Facture> ipm_statut_factures;

}

File: src/app/Models/IPM_Facture.ts
import { Employe } from "./Employe";
import { IPM_Statut_Facture } from "./IPM_Statut_Facture";
import { Prestataire } from "./Prestataire";
import { Prestation } from "./Prestations";

export class Facture{
    constructor(
        public idfacture?:number,
        // public code_acte?:string,
         public numerofacture?:number,
         public dateFacture?:Date,
         public dateSaisie?:Date,
        // public matricule?:string,
        // public montant_facture?:number,
        // public taux_ipm?:number,
         public part_ipm?:number,
         public part_patient?:number,
        // public tarification?:number,
        public ipm_prestataire?:Prestataire,
        // public ipm_prestation?:Prestation,
        // public ipm_employe?:Employe ,
        public certifier?:boolean ,
        //public ipm_details_factures?:any,
        public montantTotal?:number,
        public  ipmStatutFacture?:IPM_Statut_Facture

    ){}
}